src/types/assertion.rs

```diff
@@ -0,0 +1,661 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub struct Assertion {
+    root: Op,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(tag = "cmp")]
+#[serde(rename_all = "snake_case")]
+
+enum Comparison {
+    LessThan,
+    GreaterThan,
+    Equal,
+    NotEqual,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(tag = "header_op")]
+#[serde(rename_all = "snake_case")]
+enum HeaderOperand {
+    Literal { value: String },
+    Regex { value: String },
```

Comment:
I'm wondering if we should have a `RegexValue` that we implement the regex stuff on, so this would turn into


```suggestion
    Regex { value: RegexValue },
```

and we would have

```
type RegexOperand(String);
```

(or similar, probably need a struct to have a inner or whatever for the regex?)